### ISSUE TITLE
Questions about some key bindings

### ISSUE THREAD

**Issue by E3D3** [Alexander-Miller/treemacs#858]
My respect for making & sharing this nice program.

Treemacs works perfect for most of the things that I want to do, except:
1. The treemacs-hydra won't open when I use Emacs in a terminal. I tried it with different terminals.
2. It took a long time of experimenting to discover that the follow-mode only works within the same workspace c.q. it doesn't switch to a file in another workspace. Can this be implemented ?
3. It is confusing that the `meta`key (Alt) bindings only works when the hydra is closed. With the hydra open the `next/prev neighbour, up/down next window & root up/down` key bindings don't work and complain that the key binding is undefined.
4. Scrolling a page up & down doesn't seems to work. Navigating by going 1 line down/up is very annoying in directories with many files.
5. Treemacs makes it too easy to delete workspaces & projects so I made a backup of `~/.emacs.d/.cache/treemacs-persist`. Maybe it is nice to tell this in the documentation, have a command for this and/or an option to set a custom location for this file.

I don't know if I had the first issue before but I had issues 2, 3 & 4 also in previous Treemacs versions (< 20211007.1607).

**Comment by Alexander-Miller** [comment]
> The treemacs-hydra won't open when I use Emacs in a terminal. I tried it with different terminals.

Works for me. Do other hydras work? Does it work if you manually call `treemacs-common-helpful-hydra`?

> follow-mode only works within the same workspace

I guess you're looking for `treemacs-project-follow-mode`.

> It is confusing that the metakey (Alt) bindings only works when the hydra is closed.

No idea why that's happening. I have to investigate.

> Scrolling a page up & down doesn't seems to work.

Which keybind is that?

> Treemacs makes it too easy to delete workspaces 

I'll see about adding confirmation prompts.

**Comment by E3D3** [comment]
- Other hydra's works normally in Emacs GUI, and I just discovered that only some of them work in CLI Emacs, but none of them with the normal key bindings (I use other hydra's since this week (in this Emacs version)). `M-x treemacs-common-helpful-hydra` also didn't work in the terminal.
- Trying to use `treemacs-project-follow-mode` returns errors;
> cancel-timer: Wrong type argument: timerp, nil
> user-error: Project-Follow-Mode is only available in Emacs versions that support `window-buffer-change-functions'
- I also don't know the key bindings for scrolling page up and down but was hoping you could tell. I can't believe that everybody scrolls by going 1 line up or down c.q. pressing the same key for seconds.

**Comment by Alexander-Miller** [comment]
> I have Emacs 26.1 so can't use treemacs-project-follow-mode. 

What are you doing using such an old version? The future is now, old man.

> I also don't know the key bindings for scrolling page up and down but was hoping you could tell.

I'm the wrong person to ask. I've set up my own config to navigate 5 lines at once with J & K in evil-mode. There's also treemacs' structural keybinds to move between neighbours/parents/projects. For filter/search there's treemacs' imenu integration.

If you have a keybind that you normally use in code it's just a matter of finding out the command and binding it to treemacs' keymap.

> And the Meta key bindings M-H, M-L, M-n, M-p, M-N & M-P also work in the terminal, but still not in the GUI

Apparently that's because I added a keybind to the hydra to close with Escape. No idea what this is really about, but I'll dig deeper, and if I don't find a solution I'll get rid of the Escape bind.

**Comment by Compro-Prasad** [comment]
`q` seems better than `ESC` for closing the hydra.

**Comment by Alexander-Miller** [comment]
I've also added a confirmation for removing deleting projects (deleting workspaces already had one).

Anything that I've missed?

**Comment by E3D3** [comment]
Super !

Now my Meta key bindings works in GUI and terminal, and deleting projects need a confirmation.

Besides Christmas wrapping paper _(Dutch humor, like my remark on Clippy)_ is the only thing missing the `C-?` keybinding in the hydra, but only in the terminal. I get error:
> Error in post-command-hook (ivy--queue-exhibit): (user-error "Cannot switch buffers in a dedicated window")

 And is it normal that after updating Treemacs the icons disappear until I restart Emacs ?

**Comment by Alexander-Miller** [comment]
:+1:

**Comment by E3D3** [comment]
I have no idea when they are added but today I discovered (in the treemacs major mode help) that treemacs _has_ keybindings to scroll the treemacs window. Great ! I missed this a long time.
I think it should be also documented in the README but maybe I overlooked it or do I have to learn to inspect the major mode help & README better & again after each update (but this doesn't stimulate to update often).
The peek command is documented but it would be nice if it was added to one of the hydra's.
